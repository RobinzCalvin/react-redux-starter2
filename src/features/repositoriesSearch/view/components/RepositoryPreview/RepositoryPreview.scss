/* doiuse-disable word-break */

.repository-preview {
  padding: 1rem;
  background: white;
  color: #222;

  &__description {
    margin: 0.7rem 0 0.9rem;
  }

  &__name {
    color: #1d67e0;
    text-decoration: none;
    word-break: break-all;
    font-size: 1.4rem;
  }

  &__stars {
    display: flex;
    align-items: center;
    width: 5rem;
  }

  &__star {
    margin-right: 0.2rem;
    height: 1.3rem;
    color: #444;
    font-size: 1.2rem;
  }

  &__row {
    display: flex;
    align-items: center;
    max-width: 36rem;
  }

  &__attributes {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-top: 0.7rem;
    width: 50%;
    height: 2.35rem;
  }
}
