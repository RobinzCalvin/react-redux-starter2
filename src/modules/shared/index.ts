export { default as Layout } from './Layout/Layout';
export { default as withLayout } from './withLayout';
