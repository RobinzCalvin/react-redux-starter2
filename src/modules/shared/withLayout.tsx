import React from 'react';

import Layout from './Layout/Layout';

function withLayout(title: string) {
  return (Component: React.ComponentType) => () => (
    <Layout title={title}>
      <Component />
    </Layout>
  );
}

export default withLayout;
