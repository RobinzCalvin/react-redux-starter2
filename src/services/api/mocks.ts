import { ICakePreview } from 'shared/types/models';

// tslint:disable:max-line-length
export const cakesPreview: ICakePreview[] =  [
  {
    name: 'Vanilla Almond',
    description: 'Light and sweet but moist and rich in flavor, our Vanilla Almond cake is filled and frosted with our house vanilla buttercream. Vanilla Almond Amycakes are decorated with sparkly clear crystal sprinkles, pearl sprinkles and a ring of buttercream rosettes',
    imageURL: '',
  },
  {
    name: 'Golden Vanilla',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house vanilla buttercream. Golden Vanilla Amycakes are decorated with ascending gold sprinkles, white ganache drizzle and white chocolate curls and flakes',
    imageURL: '',
  },
  {
    name: 'Golden Chocolate',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house chocolate buttercream. Golden Chocolate Amycakes are decorated with ascending gold sprinkles, chocolate ganache drizzle and dark chocolate curls and flakes.',
    imageURL: '',
  },
  {
    name: 'Confetti',
    description: 'Moist Vanilla Almond cake studded with confetti sprinkles in the batter. Confetti Amycakes are frosted with vanilla buttercream and decorated with colorful sprinkles.',
    imageURL: '',
  },
  {
    name: 'Zebra',
    description: 'Alternating layers of vanilla almond and chocolate cakes, filled and frosted with our house vanilla buttercream.  Zebra Amycakes are decorated with chocolate ganache drizzle and white and chocolate curls. ',
    imageURL: '',
  },
];
