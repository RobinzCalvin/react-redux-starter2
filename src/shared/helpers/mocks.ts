import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export function createContainerMock() {
  return function ContainerMock() {
    return null;
  } as any;
}

export function getProfileMock(): IProfile {
  return {
    age: 100,
    nickname: 'Nickname',
    name: 'Name name',
    bio: 'profile bio',
    avatarURL: 'https://the-url.com',
  };
}

export function getGithubUserMock(): IGithubUser {
  return {
    id: 1,
    username: 'username',
    avatarURL: 'http://the-url.com',
    htmlURL: 'http://the-url.com',
  };
}

export function getDetailedGithubUserMock(): IDetailedGithubUser {
  return {
    ...getGithubUserMock(),
    followersNumber: 12,
    followingNumber: 12,
    reposNumber: 12,
    realName: 'The real name',
    location: 'Nowhere',
  };
}

export function getRepositoryMock(): IRepository {
  return {
    id: 1,
    openIssuesNumber: 1,
    starsNumber: 1,
    forksNumber: 1,
    description: 'description',
    name: 'name',
    htmlURL: 'http://the-url.com',
    language: 'en',
    updatedAt: '05/05/2005',
    owner: getGithubUserMock(),
  };
}
// TODO: моки?
