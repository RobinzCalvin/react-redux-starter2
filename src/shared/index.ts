const testContext = (require as any).context('./', true, /^.*\/(?!.*test).*\.tsx?$/);
testContext.keys().forEach(testContext);
