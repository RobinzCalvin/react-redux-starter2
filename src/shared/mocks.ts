import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export const Container: any = () => null;

export const profile: IProfile = {
  age: 100,
  nickname: 'Nickname',
  name: 'Name name',
  bio: 'profile bio',
  avatarURL: 'https://the-url.com',
};

export const githubUser: IGithubUser = {
  id: 1,
  username: 'username',
  avatarURL: 'http://the-url.com',
  htmlURL: 'http://the-url.com',
};

export const detailedGithubUser: IDetailedGithubUser = {
  ...githubUser,
  followersNumber: 12,
  followingNumber: 12,
  reposNumber: 12,
  realName: 'The real name',
  location: 'Nowhere',
};

export const repository: IRepository = {
  id: 1,
  openIssuesNumber: 1,
  starsNumber: 1,
  forksNumber: 1,
  description: 'description',
  name: 'name',
  htmlURL: 'http://the-url.com',
  language: 'en',
  updatedAt: '05/05/2005',
  owner: githubUser,
};
