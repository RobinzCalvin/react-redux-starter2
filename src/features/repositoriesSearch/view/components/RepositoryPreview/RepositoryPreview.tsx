import React from 'react';
import block from 'bem-cn';
import { bind } from 'decko';

import { IRepository } from 'shared/types/models';
import { StarIcon, Card, Link } from 'shared/view/elements';

import RepositoryAttribute from '../RepositoryAttribute/RepositoryAttribute';
import './RepositoryPreview.scss';

interface IProps {
  repository: IRepository;
  onOwnerClick(username: string): void;
}

const b = block('repository-preview');

class RepositoryPreview extends React.PureComponent<IProps> {
  public render() {
    const {
      repository: {
        name, starsNumber, htmlURL, description, updatedAt,
        owner, forksNumber, openIssuesNumber, language,
      },
    } = this.props;

    return (
      <Card>
        <div className={b()}>
          <Link className={b('name').toString()} href={htmlURL} target="_blank" rel="noopener noreferrer">
            {name}
          </Link>
          <div className={b('description')}>
            {description}
          </div>
          <div className={b('row')}>
            <div className={b('stars')}>
              <div className={b('star')}>
                <StarIcon fontSize="inherit" />
              </div>
              {starsNumber}
            </div>
            <div className={b('language')}>
              {language}
            </div>
          </div>
          <div className={b('row')}>
            <div className={b('attributes')}>
              <RepositoryAttribute
                title="Forks"
                value={forksNumber}
              />
              <RepositoryAttribute
                title="Open issues"
                value={openIssuesNumber}
              />
            </div>
            <div className={b('attributes')}>
              <RepositoryAttribute
                title="Last updated"
                value={(new Date(updatedAt)).toLocaleDateString()}
              />
              <RepositoryAttribute
                title="Owner"
                value={owner.username}
                onValueClick={this.handleOwnerClick}
                type="owner"
              />
            </div>
          </div>
        </div>
      </Card>
    );
  }

  @bind
  private handleOwnerClick() {
    const { repository: { owner: { username } }, onOwnerClick } = this.props;
    onOwnerClick(username);
  }
}

export { IProps as IRepositoryPreviewProps, RepositoryPreview };
export default RepositoryPreview;
