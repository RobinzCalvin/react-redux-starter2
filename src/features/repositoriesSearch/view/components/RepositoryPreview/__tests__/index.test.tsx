import React from 'react';
import { shallow } from 'enzyme';

import { repository } from 'shared/mocks';

import RepositoryPreview, { IRepositoryPreviewProps } from '../RepositoryPreview';

const props: IRepositoryPreviewProps = {
  repository,
  onOwnerClick: jest.fn(),
};

describe('RepositoryPreview component', () => {
  const component = shallow(<RepositoryPreview {...props} />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
