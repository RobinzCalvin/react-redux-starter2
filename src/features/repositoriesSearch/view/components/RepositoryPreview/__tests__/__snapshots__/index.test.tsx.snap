// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryPreview component should match snapshot 1`] = `
<div
  className="repository-preview"
>
  <a
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    href="http://the-url.com"
    target="_blank"
  >
    name
  </a>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    description
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        <pure(StarIcon)
          fontSize="inherit"
        />
      </div>
      1
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      en
    </div>
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <RepositoryAttribute
        title="Forks"
        value={1}
      />
      <RepositoryAttribute
        title="Open issues"
        value={1}
      />
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <RepositoryAttribute
        title="Last updated"
        value="5/4/2005"
      />
      <RepositoryAttribute
        onValueClick={[Function]}
        title="Owner"
        type="owner"
        value="username"
      />
    </div>
  </div>
</div>
`;
